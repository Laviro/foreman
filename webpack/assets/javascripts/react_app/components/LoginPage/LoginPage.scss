$caption_font_weight: 600;

$background_image: linear-gradient(
  to right,
  #005e8c,
  #084c75,
  #0a3a5e,
  #082948,
  #041a33
);

.pf-c-login {
  background-image: $background_image;

  .pf-c-login__main {
    height: fit-content;
    margin: auto 0;
  }

  svg.pf-c-spinner.pf-m-lg {
    margin-top: 14px;
  }
}
