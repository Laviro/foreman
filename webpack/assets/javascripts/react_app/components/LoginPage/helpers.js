export const adjustAlerts = alerts => {
  const submitErrors = [];
  const modifiedAlerts = [];

  alerts &&
    Object.keys(alerts).forEach((alertType, index) => {
      const alertMessage = alerts[alertType];
      if (alertType === 'error') {
        submitErrors.push(alertMessage);
      } else if (alertMessage) {
        modifiedAlerts.push({
          variant: alertType,
          title: alertMessage,
          key: index,
        });
      }
    });

  return {
    modifiedAlerts,
    submitErrors,
  };
};
