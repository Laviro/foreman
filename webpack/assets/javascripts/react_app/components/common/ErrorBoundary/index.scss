.error-boundary-foreman-app {
  .empty-state-icon {
    color: var(--pf-global--palette--red-200);
  }
}
