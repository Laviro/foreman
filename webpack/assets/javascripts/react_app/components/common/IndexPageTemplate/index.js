import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Table from '../../PF4/Table';
import PageLayout from '../../../routes/common/PageLayout/PageLayout';
import { fetchData } from '../../PF4/Table/TableActions';
import { getControllerSearchProps } from '../../../constants';
import './index.scss';
import { selectSearch, selectResponse } from '../../PF4/Table/TableSelectors';
import { getTableAPIKey } from '../../PF4/Table/TableHelpers';

const IndexPageTemplate = ({
  path,
  layoutProps,
  tableProps,
  paginationProps,
}) => {
  const dispatch = useDispatch();
  const searchQuery = useSelector(selectSearch);
  const { title, header } = useSelector(state =>
    selectResponse(state, getTableAPIKey(path))
  );
  const onSearch = nextQuery =>
    dispatch(fetchData(path, { query: nextQuery, page: 1 }));
  return (
    <PageLayout
      className="index_page_template"
      searchable
      searchQuery={searchQuery}
      searchProps={getControllerSearchProps(path)}
      onSearch={onSearch}
      header={header}
      title={title}
      {...layoutProps}
    >
      <Table
        requestPath={path}
        {...tableProps}
        paginationProps={paginationProps}
      />
    </PageLayout>
  );
};

export default IndexPageTemplate;
