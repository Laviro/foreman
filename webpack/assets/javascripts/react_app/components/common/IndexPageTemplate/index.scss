#main.index_page_template {
  padding: 0;

  h1 {
    font-weight: bold;
    margin-bottom: 0;
  }

  #index-page-top-pagination {
    margin-right: 24px;
  }

  .index-page-description {
    white-space: nowrap;
    margin-bottom: 15px;
  }
}
