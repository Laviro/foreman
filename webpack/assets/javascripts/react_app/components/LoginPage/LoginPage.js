import React from 'react';
import PropTypes from 'prop-types';
import {
  LoginPage as PFLoginPage,
  Text,
  TextContent,
  TextVariants,
} from '@patternfly/react-core';
import { translate as __, sprintf } from '../../common/I18n';
import { adjustAlerts } from './helpers';
import Alerts from './components/Alerts';
import LoginForm from './components/LoginForm';
import './LoginPage.scss';

const LoginPage = ({
  alerts,
  backgroundUrl,
  caption,
  logoSrc,
  token,
  version,
}) => {
  const { modifiedAlerts, submitErrors } = adjustAlerts(alerts);
  const textContent = (
    <TextContent>
      <Text component={TextVariants.h1} id="title">
        {__('Welcome')}
      </Text>
      {version && (
        <Text component={TextVariants.p} id="version">
          {sprintf(__('Version %s'), version)}
        </Text>
      )}
      <br />
      {caption && (
        <Text component={TextVariants.p} id="login-footer-text">
          {caption}
        </Text>
      )}
    </TextContent>
  );

  return (
    <>
      <Alerts serverAlerts={modifiedAlerts} />
      <PFLoginPage
        footerListVariants="inline"
        brandImgSrc={logoSrc}
        brandImgAlt="Foreman logo"
        backgroundImgSrc={backgroundUrl}
        backgroundImgAlt="background-image"
        textContent={textContent}
        loginTitle={__('Log in to your account')}
      >
        <LoginForm token={token} errors={submitErrors} />
      </PFLoginPage>
    </>
  );
};

LoginPage.propTypes = {
  alerts: PropTypes.shape({
    success: PropTypes.string,
    warning: PropTypes.string,
    error: PropTypes.string,
  }),
  backgroundUrl: PropTypes.string,
  caption: PropTypes.string,
  logoSrc: PropTypes.string,
  token: PropTypes.string.isRequired,
  version: PropTypes.string,
};

LoginPage.defaultProps = {
  alerts: null,
  backgroundUrl: null,
  caption: null,
  logoSrc: null,
  version: null,
};

export default LoginPage;
