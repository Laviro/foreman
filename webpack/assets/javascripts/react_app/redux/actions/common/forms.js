import { APIActions } from '../../API';

import { addToast } from '../toasts';
import { sprintf, translate as __ } from '../../../../react_app/common/I18n';

export class SubmissionError {
  constructor(errors) {
    this.errors = errors;
  }
}

const fieldErrors = ({ error }) => {
  const { errors, severity } = error;

  if (errors.base) {
    errors._error = {};
    errors._error.errorMsgs = errors.base;
    errors._error.severity = severity;
    delete errors.base;
  }

  return new SubmissionError(errors);
};

export const onError = error => {
  if (error.response.status === 422) {
    // Handle invalid form data
    throw fieldErrors(error.response.data);
  }
  throw new SubmissionError({
    _error: {
      errorMsgs: [
        `${__('Error submitting data:')} ${error.response.status} ${__(
          error.response.statusText
        )}`,
      ],
    },
  });
};

const verifyProps = (item, values) => {
  if (!item) {
    throw new Error('item must be defined, e.g. Bookmark');
  }
  if (!values) {
    throw new Error('values must be defined');
  }
};

export const submitForm = ({
  item,
  url,
  values: params,
  message,
  method = 'post',
  headers,
  apiActionTypes: actionTypes,
}) => {
  verifyProps(item, params);
  return async dispatch => {
    const uniqueAPIKey = `${item.toUpperCase()}_FORM_SUBMITTED`;

    const handleError = error => onError(error);

    const handleSuccess = ({ data }) => {
      dispatch({
        type: uniqueAPIKey,
        payload: { item, data },
      });
      dispatch(
        addToast({
          type: 'success',
          // eslint-disable-next-line no-undef
          message: message || sprintf('%s was successfully created.', __(item)),
        })
      );
    };

    dispatch(
      APIActions[method]({
        key: uniqueAPIKey,
        url,
        headers,
        params,
        actionTypes,
        handleError,
        handleSuccess,
      })
    );
  };
};
