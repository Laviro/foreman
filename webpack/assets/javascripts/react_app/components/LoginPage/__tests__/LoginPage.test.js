import React from 'react';
import LoginPage from '../LoginPage';
import { props } from '../LoginPage.fixtures';
import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

const fixtures = {
  'renders LoginPage': props,
};
describe('Login Page', () => {
  describe('rendering', () => {
    testComponentSnapshotsWithFixtures(LoginPage, fixtures);
  });

  describe('visual regression', () => {
    test('should render a Login Page', async () => {
        await render(
            <LoginPage {...props}></LoginPage>,
            { viewport: { width: 100, height: 100 } }
        );

        const screenshot = await page.screenshot();
        expect(screenshot).toMatchImageSnapshot();
    });
});
});
