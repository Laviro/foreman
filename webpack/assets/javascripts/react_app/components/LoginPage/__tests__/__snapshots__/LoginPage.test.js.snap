// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage rendering renders LoginPage 1`] = `
<Fragment>
  <Alerts
    serverAlerts={Array []}
  />
  <LoginPage
    backgroundImgAlt="background-image"
    backgroundImgSrc="/some-background"
    brandImgAlt="Foreman logo"
    brandImgSrc="/some-logo"
    footerListVariants="inline"
    loginTitle="Log in to your account"
    textContent={
      <TextContent>
        <Text
          component="h1"
          id="title"
        >
          Welcome
        </Text>
        <Text
          component="p"
          id="version"
        >
          Version %s
        </Text>
        <br />
        <Text
          component="p"
          id="login-footer-text"
        >
          some caption
        </Text>
      </TextContent>
    }
  >
    <LoginForm
      errors={
        Array [
          "some-error",
        ]
      }
      token="some token"
    />
  </LoginPage>
</Fragment>
`;
